druid/src/command.rs

```diff
@@ -228,37 +229,96 @@ pub mod sys {
 impl Selector {
     /// A selector that does nothing.
     pub const NOOP: Selector = Selector::new("");
+}
 
+impl<T> Selector<T> {
     /// Create a new `Selector` with the given string.
-    pub const fn new(s: &'static str) -> Selector {
-        Selector(s)
+    pub const fn new(s: &'static str) -> Selector<T> {
+        Selector(s, PhantomData)
+    }
+
+    pub(crate) const fn symbol(self) -> SelectorSymbol {
+        self.0
+    }
+}
+
+impl<T: Any> Selector<T> {
+    pub fn carry(self, object: T) -> Command {
+        Command::new(self, object)
     }
 }
 
 impl Command {
     /// Create a new `Command` with an argument. If you do not need
     /// an argument, `Selector` implements `Into<Command>`.
-    pub fn new(selector: Selector, arg: impl Any) -> Self {
+    pub fn new<T: Any>(selector: Selector<T>, object: T) -> Self {
         Command {
-            selector,
-            object: Some(Arc::new(arg)),
+            selector: selector.symbol(),
+            object: Arc::new(object),
         }
     }
 
     /// Used to create a command from the types sent via an `ExtEventSink`.
-    pub(crate) fn from_ext(selector: Selector, object: Option<Box<dyn Any + Send>>) -> Self {
-        let object: Option<Box<dyn Any>> = object.map(|obj| obj as Box<dyn Any>);
-        let object = object.map(|o| o.into());
-        Command { selector, object }
+    pub(crate) fn from_ext(selector: SelectorSymbol, object: Box<dyn Any>) -> Self {
+        Command {
+            selector,
+            object: object.into(),
+        }
+    }
+
+    /// Checks if this was created using `selector`.
+    pub fn is<T>(&self, selector: Selector<T>) -> bool {
+        self.selector == selector.symbol()
     }
 
-    /// Return a reference to this `Command`'s object, if it has one.
-    pub fn get_object<T: Any>(&self) -> Result<&T, ArgumentError> {
-        match self.object.as_ref() {
-            Some(o) => o.downcast_ref().ok_or(ArgumentError::IncorrectType),
-            None => Err(ArgumentError::NoArgument),
+    /// Returns `Some(reference)` to this `Command`'s object, if the selector matches.
+    ///
+    /// Returns `None` when `self.is(selector) == false`.
+    ///
+    /// If the selector has already been checked, [`get_unchecked`] can be used safely.
```

Comment:
There's not really a scenario where you would `get` and then `get_unchecked` right? So I would write:
```
Alternatively you can check the selector with [`is`] and then use [`get_unchecked`].
```